craft_providers/hookutil.py

```diff
@@ -0,0 +1,280 @@
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""Utilities for use in snap hooks.
+
+A base instance's full name may look like this:
+    base-instance-whatevercraft-buildd-base-v7-craft-com.ubuntu.cloud-buildd-daily-core24
+
+From that, the thing we care most about is the compatibility tag:
+    whatevercraft-buildd-base-v7
+"""
+
+import dataclasses
+import json
+import subprocess
+import sys
+from typing import Any
+
+from typing_extensions import Self
+
+# These are interpolated once HookHelper is instantiated with the name of the project
+_CURRENT_COMPATIBILITY_TAG = "{PROJECT_NAME}-buildd-base-v7"
+_BASE_INSTANCE_START_STRING = "base-instance"
+_CURRENT_BASE_INSTANCE_START_STRING = (
+    "{_BASE_INSTANCE_START_STRING}-{_CURRENT_COMPATIBILITY_TAG}-"
+)
+
+
+class HookError(Exception):
+    """Hook logic cannot continue.  Hooks themselves should not exit nonzero."""
+
+
+@dataclasses.dataclass
+class Instance:
+    """Represents an lxc instance."""
+
+    name: str
+    expanded_config: dict[str, str]
+
+    def base(self) -> str:
+        """Get the full name of the base instance this instance was created from."""
+        try:
+            return self.expanded_config["image.description"]
+        except KeyError as e:
+            # Unexpected, cannot continue
+            raise HookError("Could not get full base name from {self.name}") from e
+
+    def is_current_base_instance(self) -> bool:
+        """Return true if this is a base instance with the current compat tag."""
+        return self.name.startswith(_CURRENT_BASE_INSTANCE_START_STRING)
```

Comment:
If `_CURRENT_BASE_INSTANCE_START_STRING` is used only here, you could hold the current compatibility tag as an instance attribute and assemble the start string inside this method.